loader/loader_windows.c

```diff
@@ -996,4 +998,60 @@ VkResult windows_sort_physical_device_groups(struct loader_instance *inst, const
     return VK_SUCCESS;
 }
 
+char *windows_get_app_package_manifest_path(const struct loader_instance *inst)
+{
+    UINT32 numPackages = 0, bufferLength = 0;
+    PCWSTR familyName = L"Microsoft.D3DMappingLayers_8wekyb3d8bbwe";
```

Comment:
`L"Microsoft.D3DMappingLayers_8wekyb3d8bbwe";` feels like a magic constant. Is this an official 'family"? Where does the `8wekyb3d8bbwe` come from and what does it represent?

Reply:
Unfortunately, yes this is a bit of a magic constant. `"Microsoft.D3DMappingLayers_8wekyb3d8bbwe"` is the "[package family name](https://www.tmurgent.com/TmBlog/?p=3270#:~:text=Package%20Family%20Name,appears%20in%20Microsoft%20tooling%20sometimes.)" of the OpenCL™ and OpenGL® Compatibility Pack. The `8wekyb3d8bbwe` indicates that it is a package published by Microsoft - I honestly don't know exactly what it is, but it is the same suffix you'll see in the package family name for all packages we publish.

I would've preferred to use a more generic "find all packages which implement a particular feature/extension," at which point the magic constant would look a bit more like "com.microsoft.vulkan_support" but that requires C++/WinRT and has a host of other issues.

Reply:
Dang. Magic constants are no fun, but sometimes unavoidable. If you look `windows_get_registry_files()`, there are some other constants which don't have a clear meaning beyond "what windows says they should be".

Appreciate the explanation, I only ask that you summarize it and put it as a comment above the code, so that anyone coming across it understands that "this is defined by windows from the OpenGL/CL compatibility pack".

I'm actually not against adding C++ to the repo, but as i've learned, KISS goes a long way. Not adding new languages or dependencies is often a good choice, especially with so many downstream consumers.

Reply:
Yeah, the bigger problem with the C++/WinRT stuff is that it requires a WinRT initialize/uninitialize, which is a thread state that can interact poorly with other components on the thread, and also has significant problems if the loader is invoked under DllMain - which I'd hope that it never is because Windows docs do specifically say not to do things like that, but I've got battle scars from apps which do.

I'll add the comment.